RecoMTD/DetLayers/src/MTDDetLayerGeometry.cc

```diff
@@ -28,18 +28,17 @@ MTDDetLayerGeometry::MTDDetLayerGeometry() {}
 MTDDetLayerGeometry::~MTDDetLayerGeometry() {}
 
 void MTDDetLayerGeometry::buildLayers(const MTDGeometry* geo, const MTDTopology* mtopo) {
-  if (geo) {
-    // Build BTL layers
-    this->addBTLLayers(BTLDetLayerGeometryBuilder::buildLayers(*geo));
-    // Build ETL layers, depends on the scenario
-    if (mtopo) {
-      this->addETLLayers(ETLDetLayerGeometryBuilder::buildLayers(*geo, *mtopo));
-    } else {
-      LogWarning("MTDDetLayers") << "No MTD topology  is available.";
-    }
-  } else {
+  if (geo == nullptr) {
     LogWarning("MTDDetLayers") << "No MTD geometry is available.";
```

Comment:
As pointed out by the static analyzer: if either `geo` or `topo` is a `nullptr` the following builder methods will dereference a null pointer: should you more gracefully exit here, or do you rely on LogWarning + crash to identify it in the logs?

Reply:
@perrotta in principle you are correct, I might add an exception to let the code exit in a less brutal way. In practice this is such an extreme case that means something was seriously misconfigured in the job and this code should never run.
Given the fact that this issue is there since a while and this review is going already quite long, I would prefer to add such a protection to make the static analyzer happy in a send small dedicated PR, if you agree